samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/simpleforeach.cs

```diff
@@ -1,40 +1,112 @@
-        Parallel.ForEach(files, (currentFile) =>
-                                {
-                                    // The more computational work you do here, the greater
-                                    // the speedup compared to a sequential foreach loop.
-                                    string filename = Path.GetFileName(currentFile);
-                                    var bitmap = new Bitmap(currentFile);
-
-                                    bitmap.RotateFlip(RotateFlipType.Rotate180FlipNone);
-                                    bitmap.Save(Path.Combine(newDir, filename));
-
-                                    // Peek behind the scenes to see how work is parallelized.
-                                    // But be aware: Thread contention for the Console slows down parallel loops!!!
-
-                                    Console.WriteLine($"Processing {filename} on thread {Thread.CurrentThread.ManagedThreadId}");
-                                    //close lambda expression and method invocation
-                                });
-
-        // Keep the console window open in debug mode.
-        Console.WriteLine("Processing complete. Press any key to exit.");
-        Console.ReadKey();
+        static void Main(string[] args)
+        {
+            // 2 million
+            int limit = 2 * 1000000;
+            var inputs = new List<int>(limit);
+            Random radomGenerator = new Random();
+            for (int index = 0; index < limit; index++)
+            {
+                inputs.Add(radomGenerator.Next());
+            }
+
+            var watch = new Stopwatch();
+            watch.Start();
+            var primeNumbers = GetPrimeList(inputs);
+            watch.Stop();
+
+            var watchForParallel = new Stopwatch();
+            watchForParallel.Start();
+            var primeNumbersFromParallel = GetPrimeListWithParallel(inputs);
+            watchForParallel.Stop();
+
+            Console.WriteLine($"Classical For loop    | Total prime numbers : {primeNumbersFromParallel.Count} | Time Taken : {watch.ElapsedMilliseconds} ms.");
+            Console.WriteLine($"Parallel.ForEach loop | Total prime numbers : {primeNumbersFromParallel.Count} | Time Taken : {watchForParallel.ElapsedMilliseconds} ms.");
+
+            Console.WriteLine("Press any key to exit.");
+            Console.ReadLine();
+        }
+
+        /// <summary>
+        /// GetPrimeList returns Prime numbers by using sequential ForEach
+        /// </summary>
+        /// <param name="inputs"></param>
+        /// <returns></returns>
+        static IList<int> GetPrimeList(IList<int> inputs)
+        {
+            var primeNumbers = new List<int>();
+
+            foreach (var item in inputs)
+            {
+                if (IsPrime(item))
+                {
+                    primeNumbers.Add(item);
+                }
+            }
+
+            return primeNumbers;
+        }
```

Comment:
```suggestion
        static IList<int> GetPrimeList(IList<int> numbers) =>
            numbers.Where(IsPrime).ToList();
```

Use LINQ chain method, filter by `IsPrime` call `ToList`. This performs a `foreach` internally.

Reply:
Updated as suggested !!